src/Compiler.php

```diff
@@ -856,7 +981,7 @@ private static function scanFunction($input, &$name, &$arguments)
 
     private static function scanObject($input, &$object)
     {
-        if (($input === null) || ($input[0] !== Parser::TYPE_OBJECT)) {
+        if ($input === null || count($input) < 2 || $input[0] !== Parser::TYPE_OBJECT) {
```

Comment:
There is a "scanObject" function that could encapsulate this logic, so we would be safe, even if the object tokens change.